### PR TITLE
cancel-build: set status.cancelled instead of status.phase

### PR DESCRIPTION
**Pull request by twaugh**

### CODE REVIEW

**Comment by vrutkovs** [comment]
Fails here:
```
Traceback (most recent call last):
  File "/usr/bin/osbs", line 11, in <module>
    load_entry_point('osbs-client==0.33', 'console_scripts', 'osbs')()
  File "build/bdist.linux-x86_64/egg/osbs/cli/main.py", line 695, in main
  File "build/bdist.linux-x86_64/egg/osbs/cli/main.py", line 271, in cmd_cancel_build
  File "build/bdist.linux-x86_64/egg/osbs/api.py", line 49, in catch_exceptions
osbs.exceptions.OsbsException: AttributeError("'HttpSession' object has no attribute 'update'",)

Original traceback (most recent call last):
  File "build/bdist.linux-x86_64/egg/osbs/api.py", line 38, in catch_exceptions
    return func(*args, **kwargs)
  File "build/bdist.linux-x86_64/egg/osbs/api.py", line 124, in cancel_build
    response = self.os.cancel_build(build_id)
  File "build/bdist.linux-x86_64/egg/osbs/core.py", line 303, in cancel_build
    headers={"Content-Type": "application/json"})
  File "build/bdist.linux-x86_64/egg/osbs/core.py", line 181, in _update
    return self._con.update(url, headers=headers, verify_ssl=self.verify_ssl, **kwargs)
AttributeError("'HttpSession' object has no attribute 'update'",)
```

**Comment by twaugh** [comment]
I forgot I left the 'update' change in. Removed now.

**Comment by twaugh** [comment]
Could probably benefit from a unit test. @vrutkovs, I'm happy to let you continue adding commits to this branch.

File: tests/test_core.py [hunk 1]
```diff
@@ -139,9 +141,15 @@ def test_watch_build(self, openshift):
     def test_create_build(self, openshift):
         response = openshift.create_build({})
         assert response is not None
-        assert response.json()["metadata"]["name"] == TEST_BUILD
+        assert response.json()["metadata"]["name"] == TEST_CANCELLED_BUILD
```
Comment (twaugh) [inline tests/test_core.py]: Is this right?

Comment (vrutkovs) [inline tests/test_core.py]: Oops, a typo. Fixed

File: tests/build/test_build_response.py [hunk 2]
```diff
@@ -40,6 +40,23 @@ def test_get_koji_build_id(self):
...
+    def test_build_cancel(self):
+        build_response = BuildResponse({
+            'status': {
+                'phase': 'Running'
+            }
+        })
+        assert not build_response.cancelled
+        build_response.cancelled = True
+        assert build_response.cancelled
+        assert 'cancelled' in build_response.json['status'].keys()
```
Comment (twaugh) [inline tests/build/test_build_response.py]: Don't need `.keys()` here I think.

File: tests/build/test_build_response.py [hunk 3]
```diff
@@ -40,6 +40,23 @@ def test_get_koji_build_id(self):
...
+        assert build_response.json['status']['cancelled']
+        build_response.cancelled = False
+        assert not build_response.cancelled
+        assert 'cancelled' in build_response.json['status'].keys()
```
Comment (twaugh) [inline tests/build/test_build_response.py]: See above (`.keys()`).

File: tests/build/test_build_response.py (hunk 2 above)
Comment (vrutkovs) [inline tests/build/test_build_response.py]: Agreed, removed

**Comment by twaugh** [comment]
@lcarva, any objections to merging this?

**Pull request merged**